core/src/main/java/org/apache/druid/java/util/metrics/JvmMonitor.java

```diff
@@ -146,7 +151,25 @@ private void emitDirectMemMetrics(ServiceEmitter emitter)
 
   private void emitGcMetrics(ServiceEmitter emitter)
   {
-    gcCounters.emit(emitter, dimensions);
+    if (gcCounters != null) {
+      gcCounters.emit(emitter, dimensions);
+    }
+  }
+
+  @Nullable
+  private GcCounters getGcCounters()
```

Comment:
"getGcCounters" is a misleading name. I would call this method "tryCreateGcCouters".

Reply:
done